### PR TITLE
Ap 5349 build add details page

### PR DESCRIPTION
**Pull request by skdejong**
## What

[MTR2 A: Build 'Add details' page for mandatory Capital disregards](https://dsdmoj.atlassian.net/browse/AP-5349)
[MTR2 A: Add CYA, R&P and means report](https://dsdmoj.atlassian.net/browse/AP-5424)

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- The standards in the [Git Workflow document on Confluence](https://dsdmoj.atlassian.net/wiki/spaces/ATPPB/pages/4602855954/Git+Workflow) should be followed
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/controllers/providers/means/capital_disregards/mandatory_controller.rb [hunk 1]
```diff
@@ -10,7 +10,11 @@ def show
 
         def update
           @form = Providers::Means::CapitalDisregards::MandatoryForm.new(form_params)
-          render :show unless save_continue_or_draft(@form)
+
```
Comment (jsugarman) [inline app/controllers/providers/means/capital_disregards/mandatory_controller.rb:13]: You may be able to use this pattern

```ruby
ender :show unless save_continue_or_draft(@form, @legal_aid_application.next_incomplete_mandatory_disregard)
```

I believe any args are passed to go_forward anyway if form is valid

Comment (colinbruce) [reply app/controllers/providers/means/capital_disregards/mandatory_controller.rb]: You can namespace it with a key
```ruby
render :show unless save_continue_or_draft(@form, next: @legal_aid_application.next_incomplete_mandatory_disregard)
```
and then update the flow to expect arguments

File: app/controllers/providers/means/capital_disregards/add_details_controller.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+module Providers
+  module Means
+    module CapitalDisregards
+      class AddDetailsController < ProviderBaseController
+        prefix_step_with :capital_disregards
+        def show
+          @form = AddDetailsForm.new(model: capital_disregard)
```
Comment (naseberry) [inline app/controllers/providers/means/capital_disregards/add_details_controller.rb:7]: Should this meet the current pattern, that is to chain the module definitions `Providers::Means::CapitalDisregards::AddDetailsForm`?

File: app/controllers/providers/means/capital_disregards/add_details_controller.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        end
+
+        def update
+          @form = AddDetailsForm.new(form_params)
```
Comment (naseberry) [inline app/controllers/providers/means/capital_disregards/add_details_controller.rb:11]: Should this meet the current pattern, that is to chain the module definitions `Providers::Means::CapitalDisregards::AddDetailsForm`?

File: laa-apply-for-legal-aid
Comment (naseberry) [inline laa-apply-for-legal-aid:1]: What has happened here, why are we using submodules?

File: app/controllers/providers/means/capital_disregards/add_details_controller.rb (hunk 2 above)
Comment (colinbruce) [reply app/controllers/providers/means/capital_disregards/add_details_controller.rb:7]: The forms are namespace as such, but rails doesn't _need_ to chain the modules when the controller is in the same namespace.

The other two controllers are 50/50, one does, one doesn't. I think this one sets the pattern! 😁

File: laa-apply-for-legal-aid
Comment (colinbruce) [reply laa-apply-for-legal-aid:1]: This looks like an empty folder was created! :/ 
I think it can just be deleted

**Review by jsugarman — APPROVED** [review]
It looks good 👍 

Only point for possible improvement I can see would be if the controllers could be made to use the `render :show unless save_continue_or_draft(@form)` pattern but pass in the next disregard as necessary. But no worries.

I have tested both the client only bank statement and truelayer journeys plus the client bank statement with partner (with contrary interest) journey and not seen any issues, the text reflecting what i would expect 🥇

File: app/controllers/providers/means/capital_disregards/add_details_controller.rb (hunk 2 above)
Comment (skdejong) [reply app/controllers/providers/means/capital_disregards/add_details_controller.rb:7]: After a discussion today with all of the Civil Apply devs, it was decided to keep this as it is for now.

**Pull request merged**